src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -321,12 +322,23 @@ export class VSCodeNotebookController implements Disposable {
         );
     }
     private getRendererScripts(): NotebookRendererScript[] {
-        return [
-            join(this.context.extensionPath, 'out', 'datascience-ui', 'ipywidgetsKernel', 'require.js'),
-            join(this.context.extensionPath, 'out', 'ipywidgets', 'dist', 'ipywidgets.js'),
-            join(this.context.extensionPath, 'out', 'datascience-ui', 'ipywidgetsKernel', 'ipywidgetsKernel.js'),
-            join(this.context.extensionPath, 'out', 'fontAwesome', 'fontAwesomeLoader.js')
-        ].map((uri) => new NotebookRendererScript(Uri.file(uri)));
+        const scripts: string[] = [];
+        // Only used in tests & while debugging.
+        if (
+            this.context.extensionMode === ExtensionMode.Development ||
+            this.context.extensionMode === ExtensionMode.Test
+        ) {
+            scripts.push(join(this.context.extensionPath, 'out', 'datascience-ui', 'widgetTester', 'widgetTester.js'));
```

Comment:
Register the script used to query the DOM in tests, the script will check if widgets have been renderered or not & perform some basic operations.
Only included in dev & test mode.